### PR TITLE
Skip raft.db in dataonly build

### PR DESCRIPTION
**Pull request by jwilder**
###### Required for all non-trivial PRs
- [x] Rebased/mergable
- [x] Tests pass
- [x] CHANGELOG.md updated
- [x] Sign [CLA](https://influxdata.com/community/cla/) (if not already signed)

### CODE REVIEW

File: cmd/influxd/run/server_default.go [hunk 1]
```diff
@@ -129,3 +133,13 @@ func (s *Server) appendContinuousQueryService(c continuous_querier.Config) {
 	srv.QueryExecutor = s.QueryExecutor
 	s.Services = append(s.Services, srv)
 }
+
+func raftDBExists(dir string) error {
+	// Check to see if there is a raft db, if so, error out with a message
+	// to downgrade, export, and then import the meta data
+	raftFile := filepath.Join(dir, "raft.db")
+	if _, err := os.Stat(raftFile); err == nil {
```
Comment (e-dard) [inline cmd/influxd/run/server_default.go]: `os.Stat` can return errors for reasons other than the file not existing.

```go
if _, err := os.Stat(raftFile);  os.IsNotExist(err) {
    ...
}
```

File: cmd/influxd/run/server_default.go [hunk 2]
```diff
@@ -129,3 +133,13 @@ func (s *Server) appendContinuousQueryService(c continuous_querier.Config) {
...
+		return fmt.Errorf("detected %s. To proceed, you'll need to either 1) downgrade to v0.11.x, export your metadata, upgrade to the current version again, and then import the metadata or 2) delete the file, which will effectively reset your database. For more assistance with the upgrade, see: https://docs.influxdata.com/influxdb/v0.12/administration/upgrading/", raftFile)
+	}
+	return nil
```
Comment (e-dard) [inline cmd/influxd/run/server_default.go]: `return err` which will then return any errors from `os.Stat` that were not because the file didn't exist.

Comment (jwilder) [inline cmd/influxd/run/server_default.go]: I'm just moving the existing code to a new place in this PR.  Not changing the behavior.

Comment (e-dard) [inline cmd/influxd/run/server_default.go]: Well do we want to fix the code in the PR or another one? 😄

Comment (jwilder) [inline cmd/influxd/run/server_default.go]: I don't know what bug it would be fixing so I'd rather not change the behavior in this PR.

**Pull request merged**